content/en/docs/project-user-guide/image-builder/s2i-templates.md

```diff
@@ -6,4 +6,323 @@ linkTitle: "Customize S2I Templates"
 weight: 10640
 ---
 
-TBD
\ No newline at end of file
+Once you have understood the workflow and logic of Source-to-Image (S2I), you can customize Image Builder templates (i.e. S2I/B2I templates) based on your projects to extend S2I capabilities. KubeSphere provides several common Image Builder templates, such as [Python](https://github.com/kubesphere/s2i-python-container/) and [Java](https://github.com/kubesphere/s2i-java-container/).
+
+This tutorial demonstrates how to create an Image Builder that contains an Nginx service. If you need to use Runtime Image in your project, refer to [this document](https://github.com/kubesphere/s2irun/blob/master/docs/runtime_image.md) for more information about how to create a Runtime Image.
+
+## Prerequisites
+
+S2I template customization can be divided into two parts.
+
+- Part 1: S2I Image Builder customization
+  - assemble (required): the `assemble` script that builds application artifacts from source code.
+  - run (required): the `run` script that executes an application.
+  - save-artifacts (optional): the `save-artifacts` script that manages all dependencies in an incremental building process.
+  - usage (optional): the script that provides instructions.
+  - test (optional): the script for testing.
+- Part 2: definition of S2I template
+
+You need to have the required elements for S2I template customization ready in advance.
+
+{{< notice note >}}
+
+The Image Builder is compatible with that of OpenShift, and you can reuse it in KubeSphere. For more information about S2I Image Builder, refer to [S2IRun](https://github.com/kubesphere/s2irun/blob/master/docs/builder_image.md#s2i-builder-image-requirements).
+
+{{</ notice >}}
+
+## Create an Image Builder
+
+### Step 1: Prepare S2I directory
+
+1. [S2I command line tool](https://github.com/openshift/source-to-image/releases) provides an easy-to-use command to initialize a base directory structure required by the Builder. Run the following commands to install S2I CLI.
```

Comment:
[S2I command line tool](https://github.com/openshift/source-to-image/releases) provides
->
The [S2I command line tool](https://github.com/openshift/source-to-image/releases) provides

S2I CLI
->
the S2I CLI